.github/workflows/publish-c-apidocs.yml

```diff
@@ -21,27 +27,17 @@ jobs:
           sudo apt-get install libclang-cpp14
           wget https://www.doxygen.nl/files/doxygen-1.9.6.linux.bin.tar.gz
           tar xvzf doxygen-1.9.6.linux.bin.tar.gz
-      - name: Set commit ID
-        id: vars
-        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
       - name: Run doxygen
         run: |
           mkdir -p build/doxygen
           cd docs/c_cxx
           ../../doxygen-1.9.6/bin/doxygen
-      - uses: actions/checkout@v2
-        with:
-          ref: gh-pages
-          clean: false
-      - name: Move API docs into target area
-        run: |
-          rm -rf docs/api/c
-          mv build/doxygen/html docs/api/c
-      - name: Create Pull Request
-        uses: peter-evans/create-pull-request@v3
+      - name: Log source commit
+        run: git rev-parse --short HEAD > build/doxygen/html/source-version.txt
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@v4
```

Comment:
does this have any advantage over https://github.com/actions/deploy-pages?

Reply:
That one operates a little differently. It deploys the whole site from an uploaded artifact. So it would take a little more work to use. I can look into it if you think it would be better

Reply:
Actaully we might have to use the action that you mention after all due to permissions on gh-page. This is more complicated as I *think* the whole site needs to be deployed each time. Let me know if you have ideas

Reply:
I made a start on this here: https://github.com/natke/onnxruntime/blob/apidocs-gen-redux/.github/workflows/publish-java-apidocs.yml#L36 but looks like one cannot access artifacts outside of the current workflow by default. I need to work out how to do this